### PR TITLE
[LibRaw] Initial integration

### PR DESCRIPTION
**Pull request by jamiepinheiro**
Adding LibRaw fuzzing. The Google3 fork maintainers will be porting issues to GitHub.

### CODE REVIEW

**Comment by alex** [comment]
Best of luck. We test this indirectly via imagemagick, and the libraw maintainer doesn't agree with certain types of undefined behavior being treated as bugs.

**Comment by evverx** [comment]
> the libraw maintainer doesn't agree with certain types of undefined behavior being treated as bugs.

It's not the end of the world. It would probably make sense to just turn off UBSan (or a subset of the UBSan checks) then. I'd probably turn off MSan as well because I'm not exactly sure what is the point of unleashing all the sanitizers at the same time onto the project with only one maintainer.

To be fair, all the issues I was able to find where UB was mentioned were fixed except for one "left shift of negative value -1". Though I'm pretty sure that if instead of reporting the issue (with no way to reproduce it locally) someone had sent a PR fixing it it would probably have been merged.

**Comment by evverx** [comment]
FWIW since I took a look at the fuzz target, up until the point where it crashes with a stack overflow it prints a lot of the "unknown file: Unexpected end of file" lines which should probably be turned off when the code is fuzzed. Other than that, it doesn't seem to be guarded with `if (size > ... ) return 0`, which might lead to "timeout" issues that aren't really important. Plus, it looks like the fuzz target isn't linked with `clang++` somewhere causing it to fail to compile with UBsan.

**Comment by evverx** [comment]
@jamiepinheiro just to clarify, I have nothing to do with LibRaw so whatever I'm saying should be taken with a pinch of salt :-) To judge from https://github.com/LibRaw/LibRaw/issues/295#issuecomment-637304487, the LibRaw project doesn't have resources to maintain the integration and it's understandable because in its current form OSS-Fuzz requires much more manual intervention than advertised. I almost gave up after https://github.com/google/oss-fuzz/issues/3886#issuecomment-637559131 :-)

**Comment by jamiepinheiro** [comment]
@alex @evverx Thanks for the info, will make the appropriate changes!

@evverx Yeaa unfortunately looks like it might not be accepted by him, however he does seem onboard for fixing problems found. I'm hoping that this configuration for fuzzing LibRaw doesn't require more manual work since its the same fuzzer we use on it internally. I've removed UBSan and MSan, hopefully that makes it more manageable!

**Comment by evverx** [comment]
> however he does seem onboard for fixing problems found

@jamiepinheiro my understanding is that the LibRaw project would prefer to receive bug reports via GitHub and someone would have to kind of clone issues from Monorail to GitHub manually (which, in a sense, looks like a variation of https://github.com/google/oss-fuzz/issues/2985).

**Comment by jamiepinheiro** [comment]
@evvverx I see, I'll provide him with the link to track that feature request as well. It's my understanding we don't open GitHub issues since they are immediate public and pose a security risk?

**Comment by evverx** [comment]
> It's my understanding we don't open GitHub issues since they are immediate public and pose a security risk?

As far as I know, yes, the idea is to hide vulnerabilities and give maintainers some time to fix and patch them according to the standard disclosure policy. Though some project might want to chose to make some (or maybe all) bug reports public right away: https://github.com/google/oss-fuzz/issues/2582

**Comment by inferno-chromium** [comment]
> @alex @evverx Thanks for the info, will make the appropriate changes! (We also add the size check once I figure out an appropriate one)
> 
> @evverx Yeaa unfortunately looks like it might not be accepted by him, however he does seem onboard for fixing problems found. I'm hoping that this configuration for fuzzing LibRaw doesn't require more manual work since its the same fuzzer we use on it internally. I've removed UBSan and MSan, hopefully that makes it more manageable!

Lets not remove MSan and UBSan. People usually forget reenabling them and libraw has been fuzzed before indirectly, so i dont expect a whole flow of bugs.

**Comment by evverx** [comment]
> libraw has been fuzzed before indirectly, so i dont expect a whole flow of bugs.

The fuzz target crashes with a stack overflow in less than a minute so it doesn't look like the code paths it covers have been fuzzed extensively.

File: projects/libraw/project.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+homepage: "https://www.libraw.org/"
+language: c++
+primary_contact: "TODO@gmail.com"
+auto_ccs:
+    - "pinheirojamie@google.com"
+sanitizers:
+    - address
```
Comment (inferno-chromium) [inline projects/libraw/project.yaml:7]: enable memory and undefined

**Pull request merged**